src/lib/support/TestSceneTable.h

```diff
@@ -0,0 +1,140 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        https://urldefense.com/v3/__http://www.apache.org/licenses/LICENSE-2.0__;!!N30Cs7Jr!UgbMbEQ59BIK-1Xslc7QXYm0lQBh92qA3ElecRe1CF_9YhXxbwPOZa6j4plru7B7kCJ7bKQgHxgQrket3-Dnk268sIdA7Qb8$
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/clusters/scenes/SceneTableImpl.h>
+
+namespace chip {
+
+namespace SceneTesting {
+
+using FabricIndex     = chip::FabricIndex;
+using SceneTableEntry = chip::scenes::SceneTableImpl::SceneTableEntry;
+using SceneTableImpl  = chip::scenes::SceneTableImpl;
+using SceneStorageId  = chip::scenes::SceneTableImpl::SceneStorageId;
+using SceneData       = chip::scenes::SceneTableImpl::SceneData;
+
+CHIP_ERROR scene_store_test(SceneTableImpl * provider, FabricIndex fabric_index, SceneTableEntry & entry)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    SceneTableEntry temp;
+
+    LogErrorOnFailure(provider->SetSceneTableEntry(fabric_index, entry));
+    LogErrorOnFailure(provider->GetSceneTableEntry(fabric_index, entry.storageId, temp));
+    VerifyOrReturnError(temp.storageId == entry.storageId, CHIP_ERROR_WRITE_FAILED);
+    VerifyOrReturnError(temp.storageData == entry.storageData, CHIP_ERROR_WRITE_FAILED);
+
+    return err;
+}
+
```

Comment:
Please use standard testing patterns with a test under the `test/` directory of the feature, that is included in the unit tests build target, otherwise these are not run.

Reply:
Standard testing pattern implementation is ongoing, along with corner case testing for listed fabrics and no-name scene support

Reply:
TestSceneTable.cpp along with standard testing pattern